app.py

```diff
@@ -125,22 +116,33 @@ def opt_out_of_experiments():
 @app.route(f"{URL_PREFIX}/unsubscribe")
 @auth.requires_login
 def unsubscribe():
-    account_id = auth.get_account_id()
-    with DB_ENGINE.connect() as conn:
-        account_repo = DbAccountRepository(conn)
-        account_repo.remove_subscription_for_account(account_id)
-        conn.commit()
-
-    return redirect(url_for("home", error_description="Sorry to see you go. You have been unsubscribed"))
+    return render_template("pre_unsubscribe.html")
 
 
-@app.route(f"{URL_PREFIX}/email_unsubscribe/<newsletter_id>")
-def email_unsubscribe(newsletter_id):
+@app.route(f"{URL_PREFIX}/pre_unsubscribe", methods=["POST"])
+@auth.requires_login
+def pre_unsubscribe():
+    main_menu_option = request.form.get("main-menu")
+    unsubscribe_menu_option = request.form.get("unsubscribe-menu")
+    error_description = "Please reach out to poprox admins to complete this request."
     with DB_ENGINE.connect() as conn:
         account_repo = DbAccountRepository(conn)
-        account_repo.remove_subscription_for_account(auth.get_account_id())
-        conn.commit()
-    return redirect(url_for("home", error_description="Sorry to see you go. You have been unsubscribed"))
+        if main_menu_option == "leave-experiment":
+            # opt-out
+            error_description = "this is an opt out request"
+        elif unsubscribe_menu_option == "remove-email":
+            account_repo.remove_subscription_for_account(auth.get_account_id())
+            account_repo.end_consent_for_account(auth.get_account_id())
+            error_description = "You have been unsubscribed from POPROX."
+        elif unsubscribe_menu_option == "withdraw-email-poprox":
+            account_repo.remove_email_for_account(auth.get_account_id())
```

Comment:
I think this would also implicitly end consent.

Reply:
I'm a bit confused about the  option logic here, but we do have some flow graph in the IRB doc that might be helpful: https://www.figma.com/board/oiPVh9mfYUAAHQmdj31Kg3/POPROX-Unsubscribe-Prompt?node-id=0-1&t=xwMJzrMSRLuOtzEH-1

Reply:
This is an amazing graphic which I would have loved Joe to mention a month ago.

Reply:
Quick question, shall I update the current workflow with this one? Do we have a variant number somewhere or can I ignore that for now?

Reply:
Variant number would be a seperate follow-up task.
It is documented and is 100% placebo effect.

Reply:
I don't think we need to make the system work _exactly_ like this supposes, mostly because I don't think we need to do this with 4 or 5 binary decision points. I think we need to ask to what degree does _our_ decision workflow achive equal outcomes to _this_ workflow
(although I recognize that I'm posting this 4 days too late :| )

Reply:
So far, I have added language that skips the "variant" part. I could add it based on an additional tiny PR once we know where that number is coming from

Reply:
So is this basically ready for final re-review then? I might just want to spin-it-up to _see_ the new workflow since the front-end code is looking a bit more involved than last time.

Reply:
Yes, I think this is ready for final re-review. Please let me know when you test the workflow. I am skeptical about this working 😶.